src/ui/public/vis/request_handlers/courier.js

```diff
@@ -100,7 +90,7 @@ const CourierRequestHandlerProvider = function (Private, courier, timefilter) {
             searchSource.lastQuery = {
               filter: _.cloneDeep(searchSource.get('filter')),
               query: _.cloneDeep(searchSource.get('query')),
-              aggs: _.cloneDeep(copyAggs(vis.aggs.getRequestAggs())),
+              aggs: calculateObjectHash(vis.aggs.getRequestAggs()),
```

Comment:
I think we should change the name of this variable now to aggsHash or something. Otherwise it kinda looks like a bug above, comparing the object hash to the last query aggs.